gmagickpixel_methods.c

```diff
@@ -258,3 +258,125 @@ PHP_METHOD(gmagickpixel, setcolorvalue)
 	GMAGICK_CHAIN_METHOD;
 }
 /* }}} */
+
+/* {{{ proto float GmagickPixel::getColorValueQuantum(int color )
+	Gets the quantum color of the GmagickPixel.
+*/
+PHP_METHOD(gmagickpixel, getcolorvaluequantum)
+{
+	php_gmagickpixel_object *internp;
+	zend_long colorquantum;
+	double color_value_quantum = 0;
+
+	/* Parse parameters given to function */
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "l", &colorquantum) == FAILURE) {
+		return;
+	}
+
+	internp = Z_GMAGICKPIXEL_OBJ_P(getThis());
+
+	switch (colorquantum) {
+
+		case GMAGICK_COLOR_BLACK:
+			color_value_quantum = PixelGetBlackQuantum(internp->pixel_wand);
+			break;
+
+		case GMAGICK_COLOR_BLUE:
+			color_value_quantum = PixelGetBlueQuantum(internp->pixel_wand);
+			break;
+
+		case GMAGICK_COLOR_CYAN:
+			color_value_quantum = PixelGetCyanQuantum(internp->pixel_wand);
+			break;
+
+		case GMAGICK_COLOR_GREEN:
+			color_value_quantum = PixelGetGreenQuantum(internp->pixel_wand);
+			break;
+
+		case GMAGICK_COLOR_RED:
+			color_value_quantum = PixelGetRedQuantum(internp->pixel_wand);
+			break;
+
+		case GMAGICK_COLOR_YELLOW:
+			color_value_quantum = PixelGetYellowQuantum(internp->pixel_wand);
+			break;
+
+		case GMAGICK_COLOR_MAGENTA:
+			color_value_quantum = PixelGetMagentaQuantum(internp->pixel_wand);
+			break;
+
+		case GMAGICK_COLOR_OPACITY:
+			color_value_quantum = PixelGetOpacityQuantum(internp->pixel_wand);
+			break;
+
+		default:
+			GMAGICK_THROW_GMAGICKPIXEL_EXCEPTION(internp->pixel_wand, "Unknown color type");
+			break;
+	}
+	RETVAL_DOUBLE(color_value_quantum);
```

Comment:
This should be returning a RETVAL_LONG - I think. For whatever reason my computers decided to refuse to compile GraphicsMagick today. But looking in GraphicsMagick in magick/image.h - all of the types that Quantum can be are ints.